march_hardware/src/IMotionCube.cpp

```diff
@@ -35,6 +43,28 @@ IMotionCube::IMotionCube(int slave_index, std::unique_ptr<AbsoluteEncoder> absol
   // Multiply by ten to ensure the rotational joints keep using absolute encoders. These are somehow more accurate
   // even though they theoretically shouldn't be.
 }
+IMotionCube::IMotionCube(int slave_index, std::unique_ptr<AbsoluteEncoder> absolute_encoder,
+                         std::unique_ptr<IncrementalEncoder> incremental_encoder, std::stringstream& sw_stream,
+                         ActuationMode actuation_mode)
+  : Slave(slave_index)
+  , absolute_encoder_(std::move(absolute_encoder))
+  , incremental_encoder_(std::move(incremental_encoder))
+  , sw_stream_(std::move(sw_stream))
+  , actuation_mode_(actuation_mode)
+{
+  if (!this->absolute_encoder_ || !this->incremental_encoder_)
+  {
+    throw std::invalid_argument("Incremental or absolute encoder cannot be nullptr");
+  }
+  ROS_INFO("length in constructor: %d", this->sw_stream_.str().length());
+  this->absolute_encoder_->setSlaveIndex(slave_index);
+  this->incremental_encoder_->setSlaveIndex(slave_index);
+  this->is_incremental_more_precise_ =
+      (this->incremental_encoder_->getTotalPositions() * this->incremental_encoder_->getTransmission() >
+       this->absolute_encoder_->getTotalPositions() * 10);
+  // Multiply by ten to ensure the rotational joints keep using absolute encoders. These are somehow more accurate
+  // even though they theoretically shouldn't be.
+}
```

Comment:
To avoid code redundancy, you can call the other constructor from the initializer list.
```suggestion
  : IMotionCube(slave_index, std::move(absolute_encoder), std::move(incremental_encoder), actuation_mode)
{
  this->sw_stream_ = std::move(sw_stream);
  ROS_INFO("length in constructor: %lu", this->sw_stream_.str().length());
}
```

Reply:
could you please elaborate, either here or via Slack. I do not fully understand what you want changed even though you have provided a suggestion.